src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,318 @@
+                if (stream == null)
+                    return null;
+                streams[node.Name!] = stream;
+            }
+
+            return streams;
+        }
+
+        /// <summary>
+        /// Asynchronously returns a stream of <see cref="ExecutionResult"/> responses for the
+        /// specified <see cref="ExecutionNode"/>.
+        /// </summary>
+        protected virtual async Task<IObservable<ExecutionResult>?> ResolveEventStreamAsync(ExecutionContext context, ExecutionNode node)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var resolveContext = new ReadonlyResolveFieldContext(node, context);
+
+            IObservable<object?> subscription;
+
+            try
+            {
+                if (node.FieldDefinition?.Subscriber == null)
+                {
+                    // todo: this should be caught by schema validation
+                    throw new InvalidOperationException($"Subscriber not set for field '{node.Field.Name}'.");
+                }
+
+                subscription = await node.FieldDefinition.Subscriber.SubscribeAsync(resolveContext).ConfigureAwait(false);
+
+                if (subscription == null)
+                {
+                    throw new InvalidOperationException($"No event stream returned for field '{node.Field.Name}'.");
+                }
+            }
+            catch (OperationCanceledException) when (context.CancellationToken.IsCancellationRequested)
+            {
+                throw;
+            }
+            catch (ExecutionError error)
+            {
+                error.Path = node.ResponsePath;
+                error.AddLocation(node.Field, context.Document);
+                context.Errors.Add(error);
+                return null;
+            }
+            catch (Exception exception)
+            {
+                context.Errors.Add(await HandleExceptionInternalAsync(context, node, exception,
+                    $"Could not subscribe to field '{node.Field.Name}'.").ConfigureAwait(false));
+                return null;
+            }
+
+            // cannot throw an exception here
+            return subscription
+                .SelectCatchAsync(
+                    async (value, token) =>
+                    {
+                        // duplicate context to prevent multiple event streams from sharing the same context,
+                        // and clear errors/metrics/extensions. Free array pool leased arrays after execution.
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
```

Comment:
May be moved into local function if you want.

Reply:
Sure